modulemd/v2/modulemd-defaults.c

```diff
@@ -44,6 +45,31 @@ enum
 static GParamSpec *properties[N_PROPS];
 
 
+gboolean
+modulemd_defaults_equals (ModulemdDefaults *self_1, ModulemdDefaults *self_2)
```

Comment:
This is the parent implementation, but if you name it as the public name, you'll never call the `equals()` routine on the child classes.

```suggestion
modulemd_defaults_default_equals (ModulemdDefaults *self_1, ModulemdDefaults *self_2)
```

You need to implement a proper `modulemd_defaults_equals()` that calls this function via `klass->equals()`.